sdk/communication/AzureCommunicationChat/Source/ChatClient.swift

```diff
@@ -167,4 +185,37 @@ public class ChatClient {
             }
         }
     }
+
+    /// Start receiving realtime notifications.
+    /// Call this function before subscribing to any event.
+    public func startRealTimeNotifications () {
+        if self.signalingClient == nil {
+            print("no signaling client is initialized")
+        }
+        if self.isRealtimeNotificationsStarted {
+            return
+        }
+
+        self.isRealtimeNotificationsStarted = true
+        self.signalingClient?.start()
+    }
+
+    /// Stop receiving realtime notifications.
+    /// This function would unsubscribe to all events.
+    public func stopRealTimeNotifications () {
+        if self.signalingClient == nil {
+            print("no signaling client is initialized")
+        }
+        self.isRealtimeNotificationsStarted = false
+        self.signalingClient?.stop()
+    }
+
+    /// Subscribe to chat events
+    public func on (event: String, listener:  @escaping EventListener) {
+        guard let _ =  ChatEventId(rawValue: event) else {
+            print("the event id provided is not supported")
+            return
+        }
```

Comment:
This should not be a print statement. It should probably invoke the client logger to log an error.